src/cim/components/ClusterDeployment/ClusterDeploymentWizard.tsx

```diff
@@ -33,18 +56,48 @@ const ClusterDeploymentWizard: React.FC<ClusterDeploymentWizardProps> = ({
 
     switch (stepId) {
       case 'hosts-selection':
-        return agentClusterInstall?.metadata?.name ? (
-          <ClusterDeploymentHostSelectionStep
-            clusterDeployment={clusterDeployment}
-            agentClusterInstall={agentClusterInstall}
-            onClose={onClose}
-            onSaveHostsSelection={onSaveHostsSelection}
-            agents={agents}
-            aiConfigMap={aiConfigMap}
-          />
-        ) : (
-          <LoadingState />
-        );
+        if (agentClusterInstall?.metadata?.name) {
```

Comment:
Instead of defining conditional logic under hosts-selection step, could we define a new `host-discovery` step and use different steps for AI and CIM flow? IMHO it would greatly simplify the wizard here.

Ideally the wizard would just return components for steps and any specific logic should be handled inside the component (loading state which we have here originally as well)

Reply:
Sure. To do that, we will need to change logic around `next` in the `ClusterDeploymentDetailsStep` and `onBack` in the `ClusterDeploymentNetworkingStep`

Reply:
done